redfish-core/lib/environment_metrics.hpp

```diff
@@ -117,58 +115,100 @@ inline void
 inline void getPowerWatts(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                           const std::string& chassisID)
 {
-    const std::array<const char*, 1> totalPowerInterfaces = {
+    const std::string sensorPath = "/xyz/openbmc_project/sensors";
+    const std::array<std::string, 1> sensorInterfaces = {
         "xyz.openbmc_project.Sensor.Value"};
-    const std::string& totalPowerPath =
-        "/xyz/openbmc_project/sensors/power/total_power";
     crow::connections::systemBus->async_method_call(
-        [asyncResp, chassisID, totalPowerPath](
-            const boost::system::error_code ec,
-            const std::vector<std::pair<std::string, std::vector<std::string>>>&
-                object) {
+        [asyncResp, chassisID](const boost::system::error_code ec,
+                               const std::vector<std::string>& paths) {
             if (ec)
             {
                 BMCWEB_LOG_DEBUG << "DBUS response error";
                 messages::internalError(asyncResp->res);
                 return;
             }
-            for (const auto& tempObject : object)
+            bool judgment = false;
+            for (const auto& tempPath : paths)
             {
-                const std::string& connectionName = tempObject.first;
+                sdbusplus::message::object_path path(tempPath);
+                std::string leaf = path.filename();
+                if (!leaf.compare("total_power"))
+                {
+                    judgment = true;
+                    break;
+                }
+            }
+            if (judgment)
+            {
+                const std::array<const char*, 1> totalPowerInterfaces = {
+                    "xyz.openbmc_project.Sensor.Value"};
+                const std::string& totalPowerPath =
+                    "/xyz/openbmc_project/sensors/power/total_power";
```

Comment:
We should get "totalPowerPath" from the above call instead of hardcoding it here.

Reply:
Something like:
insert at line 130:   const std::string& totalPowerPath = "";
...
insert at line 137: totalPowerPath = path;